base/serialize.jl

```diff
@@ -770,11 +835,48 @@ function deserialize(s::AbstractSerializer, ::Type{GlobalRef})
     end
 end
 
+function deserialize_wrapper_datatype(s::AbstractSerializer)
+    number = read(s.io, UInt64)
+    if haskey(known_object_data, number)
+        ndt = known_object_data[number]::DataType
+        makenew = false
+    else
+        ndt = ccall(:jl_new_uninitialized_datatype, Any, ())::DataType
+        makenew = true
+        ndt.super = Any
+    end
+    resolve_ref_immediately(s, ndt)
+
+    # deserialize_typename can assume unwrap_unionall(wrapper).parameters is set
```

Comment:
doesn't it also assume that the layout, mutabl, and types fields are valid, and may start propagating bad assumption about `ndt` as soon as the `ccall` returned the incomplete object?